rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java

```diff
@@ -166,8 +167,12 @@ protected boolean containsPK() {
         if (b != 0) {
             throw new NotSupportYetException("not support sql [" + recognizer.getOriginalSQL() + "]");
         }
+        // pk auto generated while single insert primary key is expression
+        if (pkValues.size() == 1 && pkValues.get(0) instanceof SqlMethodExpr) {
+            pkValues = getPkValuesByAuto();
+        }
         // pk auto generated while column exists and value is null
-        if (pkValues.size() > 0 && pkValues.get(0) instanceof Null) {
+        else if (pkValues.size() > 0 && pkValues.get(0) instanceof Null) {
```

Comment:
pkValues.size() > 0 && pkValues.stream().allMatch(pkValue -> pkValue instanceof Null)
easy to understand